### PR TITLE
Prepare for CakePHP 3.6

### PR DESCRIPTION
**Pull request by felixkempf**
Make it ready for CakePHP 3.6 by removing deprecated function calls

Also refactors some minor control flows and code style

### CODE REVIEW

File: src/Controller/Component/MaintenanceComponent.php [hunk 1]
```diff
@@ -99,15 +102,15 @@ public function beforeFilter()
...
                 echo 'Unauthorized';
                 exit;
-            } else {
-                if ($_SERVER['PHP_AUTH_USER'] == $user && $_SERVER['PHP_AUTH_PW'] == $password) {
-                    $this->_controller->Cookie->write($cookieName, true);
+            }
 
-                    return $this->_controller->redirect($successUrl);
-                }
+            if ($_SERVER['PHP_AUTH_USER'] == $user && $_SERVER['PHP_AUTH_PW'] == $password) {
```
Comment (cleptric) [inline src/Controller/Component/MaintenanceComponent.php]: There should be APIs from Cake to achieve that.

File: src/Controller/Component/MaintenanceComponent.php [hunk 2]
```diff
@@ -28,7 +29,7 @@ public function __construct(ComponentRegistry $registry, array $config = [])
     {
         parent::__construct($registry, $config);
 
-        if ($registry->getController() instanceof \Cake\Controller\Controller) {
+        if ($registry->getController() instanceof Controller) {
             $this->_controller = $registry->getController();
```
Comment (cleptric) [inline src/Controller/Component/MaintenanceComponent.php]: `$this->getController()` (Which is a wrapper for the same code above)

File: src/Controller/Component/MaintenanceComponent.php [hunk 3]
```diff
@@ -99,15 +102,15 @@ public function beforeFilter()
...
                 exit;
-            } else {
-                if ($_SERVER['PHP_AUTH_USER'] == $user && $_SERVER['PHP_AUTH_PW'] == $password) {
-                    $this->_controller->Cookie->write($cookieName, true);
+            }
 
-                    return $this->_controller->redirect($successUrl);
-                }
+            if ($_SERVER['PHP_AUTH_USER'] == $user && $_SERVER['PHP_AUTH_PW'] == $password) {
+                $this->_controller->Cookie->write($cookieName, true);
```
Comment (cleptric) [inline src/Controller/Component/MaintenanceComponent.php:107]: Same here

**Comment by cleptric** [comment]
Besides some minor things, this looks great. We should release it as a new major.

File: src/Controller/Component/MaintenanceComponent.php (hunk 1 above)
Comment (felixkempf) [reply src/Controller/Component/MaintenanceComponent.php]: `BasicAuthenticate` uses these headers, but is out of the scope of this Pull Request.

But let's try this new feature of github (added type safety):

```suggestion
            if ($_SERVER['PHP_AUTH_USER'] === $user && $_SERVER['PHP_AUTH_PW'] === $password) {
```

File: src/View/Helper/MenuHelper.php [hunk 4]
```diff
@@ -66,7 +67,13 @@ class MenuHelper extends Helper
...
     {
         parent::__construct($View, $config);
-        $this->_currentUrl = Router::parse(Router::url());
+        $this->_currentUrl = [
+            'controller' => $this->request->getParam('controller'),
+            'action' => $this->request->getParam('action'),
+            'pass' => $this->request->getParam('pass'),
+            'prefix' => $this->request->getParam('prefix'),
+            'plugin' => $this->request->getParam('plugin'),
+        ];
```
Comment (felixkempf) [inline src/View/Helper/MenuHelper.php:76]: Use `Request::getAttribute('params')` instead.

**Pull request merged**